core/src/engine.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "knoting/engine.h"
+
+namespace knot {
+
+Engine::Engine() {
+    m_windowModule = std::make_shared<knot::Window>(m_windowWidth, m_windowHeight, m_windowTitle);
+
+    m_engineModules.emplace_back(m_windowModule);
+
```

Comment:
When other subsystems are created are they all being added here?
Is there any instance where modules would be added at runtime?

Reply:
I would say that that would rarely be the case, since we will be exposing the EnTT library (the library which manages the ECS). So we can add other systems from outside the core library by retrieving the registry and directly making a system.